ToDoApp/ToDoApp/ToDoService.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  ToDoDetail.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-11-29.
+//
+
+import Foundation
+
+enum TaskError: Error {
+    case unavailableTask
+}
+
+struct Task {
+    var id: String = ""
+    var name: String = ""
+    var deadline: Date = Date.now
+    var status: Bool = false
```

Comment:
do we really wand id to be a variable? should we ever change it?
do we really want to create a Task with a default id value of empty string? I suggest to use UUID type to represent Id of a task
do we really want to create a Task without a name with default value of empty string? 
since deadline could be optional shouldn't it's default value be nil?

Reply:
Oh I thought if a task without any deadline, it'll be set for today as the default deadline. I will change it to nil now.
And for a task without a name, I think I'll put the default value named "New Task." I'm fixing it now.

Reply:
But why do we need a default string for the name? A task will be always created with a name provided by the user
I also believe that the note should be either required to be provided by the user or be optional since I don't see a case when the user would like to always provide a note. for example what would be the note if the task is just to by a coffee

here is an example: the id is let, since it will never change, for title there is no default value, since the user will always provide own value, note is optional since it is not required.
```
struct Task {
     let id: UUID = UUID()
     var title: String
     var note: String?
     var deadline: Date?
     var status: Bool = false
 }
```

The last thing is the instantiation of the id. It would be better to carry over the responsibility of creating the id to another object, to be able to properly test task creation, but I think we can skip this for now